assets/model_monitoring/components/src/shared_utilities/span_tree_utils.py

```diff
@@ -0,0 +1,242 @@
+        """Get the span id."""
+        return self.get_node_attribute("span_id")  # type: ignore
+
+    @property
+    def parent_id(self) -> str:
+        """Get the span's parent id."""
+        return self.get_node_attribute("parent_id")  # type: ignore
+
+    @property
+    def children(self) -> List["SpanTreeNode"]:
+        """Get the span's children as list."""
+        return self._children
+
+    @children.setter
+    def children(self, value: list) -> None:
+        """Set the span's children."""
+        self._children = value
+
+    @property
+    def span_type(self) -> str:
+        """Get the span's type."""
+        return self.get_node_attribute("span_type")  # type: ignore
+
+    @property
+    def start_time(self) -> datetime:
+        """Get the span's start_time."""
+        return self.get_node_attribute("start_time")  # type: ignore
+
+    @property
+    def end_time(self) -> datetime:
+        """Get the span's end_time."""
+        return self.get_node_attribute("end_time")  # type: ignore
+
+    @property
+    def attributes(self) -> str:
+        """Get the span's attributes."""
+        return self.get_node_attribute("attributes")  # type: ignore
+
+    @property
+    def input(self) -> str:
+        """Get the span's input."""
+        return self.get_node_attribute("input")  # type: ignore
+
+    @property
+    def output(self) -> str:
+        """Get the span's output."""
+        return self.get_node_attribute("output")  # type: ignore
+
+    @property
+    def status(self) -> str:
+        """Get the span's status."""
+        return self.get_node_attribute("status")  # type: ignore
+
+    @property
+    def framework(self) -> str:
+        """Get the span's framework."""
+        return self.get_node_attribute("framework")  # type: ignore
+
+    @property
+    def name(self) -> str:
+        """Get the span's name."""
+        return self.get_node_attribute("name")  # type: ignore
+
+    @property
+    def trace_id(self) -> str:
+        """Get the span's trace id."""
+        return self.get_node_attribute("trace_id")  # type: ignore
+
+    @classmethod
+    def create_node_from_dict(cls, span_node_dict: dict) -> "SpanTreeNode":
+        """Parse dict representation to create a single SpanTree node."""
+        if span_node_dict is None or span_node_dict == {}:
+            raise InvalidInputError(
+                "Can not create SpanTreeNode from empty root_span." +
+                f" Input encountered: '{span_node_dict}'."
+            )
+
+        span_node_dict['start_time'] = datetime.fromisoformat(span_node_dict['start_time'])
+        span_node_dict['end_time'] = datetime.fromisoformat(span_node_dict['end_time'])
+        children_dicts = span_node_dict.pop('children', [])
+
+        obj = cls.__new__(cls)
+        super(SpanTreeNode, obj).__init__()
+
+        child_nodes = []
+        for child_dict in children_dicts:
+            new_node = SpanTreeNode.create_node_from_dict(child_dict)
```

Comment:
when I test, I found this child_node is a str. I think we need to do json.loads(child_dict) here. Can you double check?

Reply:
I changed how the SpanTree outputs to json_string and loads from json_string. Yuan's issue happened with the old preprocessor output format